src/controller/CHIPDeviceController.cpp

```diff
@@ -753,7 +753,10 @@ CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, RendezvousParam
         }
     }
 #endif
-    mRendezvousSession->OnRendezvousConnectionOpened();
+    exchangeCtxt = mExchangeMgr->NewContext(SecureSessionHandle(), &mPairingSession);
+    VerifyOrExit(exchangeCtxt != nullptr, err = CHIP_ERROR_INTERNAL);
+
+    err = mPairingSession.Pair(params.GetPeerAddress(), params.GetSetupPINCode(), mNextKeyId++, exchangeCtxt, this);
```

Comment:
nit: I wonder if we should increment `mNextKeyID` above this line. Just to make it clearer that mNextKey changes here. It's probably fine either way.

Reply:
Do you mean after this line? No strong opinion.. We can probably do that.